src/main/kotlin/ru/hse/dynamomock/service/AWSDynamoDBMockService.kt

```diff
@@ -208,13 +215,96 @@ class AWSDynamoDBMockService(private val storage: DataStorageLayer) {
+                    throw AWSMockCSVException("Wrong value format. Use <column_name>|<type>")
+                }
+                val columnName = value[0].trim()
+                val type = value[1].trim()
+
+                if (!allowedTypes.contains(DynamoType.valueOf(type))) {
+                    throw AWSMockCSVException("Function scanItems supports only S, N, NS, SS, NULL, BOOL types right now")
+                }
+                columnsInfo.add(columnName to type)
+            }
+        }
+
+        var i = 0
+        val itemsList = mutableListOf<Map<String, AttributeValue>>()
+        File(fileName).forEachLine {
+            if (i != 0) {
+                val columnValues = it.split(";").map { v -> v.trim() }
+                if (columnValues.size != columnsInfo.size) {
+                    throw AWSMockCSVException("${i + 1} row's size doesn't match the size of first row")
+                }
+                val item = columnValues.mapIndexed { index, element ->
+                    val (columnName, type) = columnsInfo[index]
+                    columnName to when (DynamoType.valueOf(type)) {
+                        S -> AttributeValue.builder().s(element).build()
+                        N -> AttributeValue.builder().n(element).build()
+                        NS -> {
+                            val list = element.split(",")
+                            AttributeValue.builder().ns(list).build()
+                        }
+                        SS -> {
+                            val list = element.split(",")
+                            AttributeValue.builder().ss(list).build()
+                        }
+                        NULL -> {
+                            AttributeValue.builder().nul(element == "true").build()
+                        }
+                        BOOL -> {
+                            AttributeValue.builder().bool(element == "true").build()
+                        }
+                        L -> {
+                            val attributeValues = Json.decodeFromString<List<AttributeTypeInfo>>(element)
+                            AttributeValue.builder()
+                                .l(attributeValues.map { v -> toAttributeValue(v) })
+                                .build()
+                        }
+                        M -> {
+                            val attributeValues = Json.decodeFromString<Map<String, AttributeTypeInfo>>(element)
+                            AttributeValue.builder().m(
+                                attributeValues.map { v -> v.key to toAttributeValue(v.value) }.toMap()
+                            ).build()
+                        }
+                        else -> throw AWSMockCSVException("Function scanItems supports only S, N, NS, SS, NULL, BOOL types right now")
+                    }
+                }.toMap()
+
+                itemsList.add(item)
+            }
+            i++
+        }
+        itemsList.forEach {
```

Comment:
Why do we do an actual insertion in a separate loop, instead of a file line iteration loop?